### ISSUE TITLE
Explain why install as System Addon as part of Fx CI is weird

### ISSUE THREAD

**Issue by gregglind** [mozilla/shield-studies-addon-utils#233]
@biancadanforth

**Comment by biancadanforth** [comment]
# TL;DR
Installing a shield study add-on as a system add-on is non-trivial, and there are behavioral differences between how system add-ons and add-ons associated with a user profile behave.
Note: This experience was using v4 StudyUtils with a boostrapped legacy add-on circa March 2018.
[Reference](https://github.com/biancadanforth/tracking-protection-shield-study/pull/12)

## Installing a shield study add-on as a system add-on is non-trivial
* Need to create custom `jar.mn` and `moz.build` files, which are only used for installing as a system add-on and are hard to auto-generate.
* Need to modify `./browser/extensions/moz.build`, which is hard to automate.
* Need to copy over add-on source files (not an XPI) into appropriate directory in `mozilla-unified`.
  * This was the only part I was easily able to script so far.
* Need to modify file structure (and all paths within the add-on code) to match the assumptions of `./mach build`

## There are differences between how system add-ons and add-ons associated with a user profile behave.
* The lifecycle event `ADDON_INSTALL` never fires, since the add-on is installed at compile time.
  * Most Shield studies have different behavior on add-on install.
* The add-on is not accessible via `about:addons` for visibility
* You can’t uninstall the add-on and test those behaviors as a system add-on.
* @rhelmer might have more to say about other differences.

**Comment by rhelmer** [comment]
I think we don't need this anymore, now that the Mozilla test runners support loading a XPI as part of running CI (this can be done via a try-server push)
